plugins/http_plugin/http_plugin.cpp

```diff
@@ -353,7 +431,14 @@ namespace eosio {
             }
          }
 
-         if( options.count( "https-server-address" ) && options.at( "https-server-address" ).as<string>().length()) {
+         if( options.count( my->unix_socket_path_option_name ) && !options.at( my->unix_socket_path_option_name ).as<boost::filesystem::path>().empty()) {
+            boost::filesystem::path sock_path = options.at(my->unix_socket_path_option_name).as<boost::filesystem::path>();
+            if (sock_path.is_relative())
+               sock_path = app().data_dir() / sock_path;
+            my->unix_endpoint = asio::local::stream_protocol::endpoint(sock_path.string());
```

Comment:
Does boost do something sane with permissions on the socket file?

Reply:
It just will get the default via whatever current umask is. You'll find in the local_endpoint.hpp file I monkey with the umask to ensure the file is created 0660